File: fsharp/ppm.fs
open System

//Defined Presets
let version = "P3"  // P3 for readable text, P6 for binary
let res = "256 256" // image width and height
let maxVal = "255"  // max color value
let Color = {red: System.int; green: System.int; blue: System.int;}

//Red Green Gredient
let rec makePPM color =
	if (color.red = 255) then color
	elif (color.green = 255) then color

	printfn "%s %s %s" color.red color.green color.blue

	(makePPM color.green + 1)
	(makePPM color.red + 1)

colorBlack = {red: 0; green: 0; blue: 0;}

makePPM colorBlack

File: Python/ppm.py
#PPM Header
width  = 256
height = 256
maxval = 255
header = 'P3\n' + str(width) + ' ' + str(height) + '\n' + str(maxval) + '\n'

# Data Array
img = []

#generation function for red and green:
def generatePPMRG():
    for x in range(0, height):
        for y in range(0, width):
            temp = (x, y, 0)
            img.append(temp)

#generation function for red and blue:
def generatePPMRB():
    for x in range(0, height):
        for y in range(0, width):
            temp = (x, 0, y)
            img.append(temp)

#generation function for green and blue:
def generatePPMGB():
    for x in range(0, height):
        for y in range(0, width):
            temp = (0, x, y)
            img.append(temp)

#generation function for green and red:
def generatePPMGR():
    for x in range(0, height):
        for y in range(0, width):
            temp = (y, x, 0)
            img.append(temp)

#generation function for blue and red:
def generatePPMBR():
    for x in range(0, height):
        for y in range(0, width):
            temp = (y, 0, x)
            img.append(temp)

#generation function for blue and green:
def generatePPMBG():
    for x in range(0, height):
        for y in range(0, width):
            temp = (0, x, y)
            img.append(temp)

#function call. make a switch later
generatePPMRG()

#write image to file
with open("out.ppm", 'w') as f:
    f.write(header)
    for x in range(0,len(img)):
        r,g,b = img[x]
        f.write(str(r) + ' ' + str(g) + ' ' + str(b) + '\n')
